### PR TITLE
[sgd/v1] Add API annotations

### PR DESCRIPTION
**Pull request by krfricke**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/sgd/tf/tf_runner.py [hunk 1]
```diff
@@ -14,6 +15,7 @@ def _try_import_strategy():
     return tf.distribute.experimental.MultiWorkerMirroredStrategy
 
 
+@DeveloperAPI
```
Comment (amogkam) [inline python/ray/util/sgd/tf/tf_runner.py:18]: I don't think we need annotations for this right?

```
Developer APIs are lower-level methods explicitly exposed to advanced Ray
    users and library developers. Their interfaces may change across minor
    Ray releases.
```

`TFRunner` is purely internal and not even exposed to advanced users.

File: python/ray/util/sgd/torch/torch_runner.py [hunk 2]
```diff
@@ -26,6 +27,7 @@
     pass
 
 
+@PublicAPI(stability="beta")
```
Comment (amogkam) [inline python/ray/util/sgd/torch/torch_runner.py:30]: See comment above, we shouldn't need any annotations for this.

File: python/ray/util/sgd/torch/worker_group.py [hunk 3]
```diff
@@ -20,6 +21,7 @@
 logger = logging.getLogger(__name__)
 
 
+@DeveloperAPI
```
Comment (amogkam) [inline python/ray/util/sgd/torch/worker_group.py:24]: Same as above, we shouldn't need any annotations for this. This is purely internal.

**Review by amogkam — CHANGES_REQUESTED** [review]

File: python/ray/util/sgd/data/dataset.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
+from ray.util.annotations import PublicAPI
 from ray.util.iter import ParallelIterator, from_iterators
 
 
+@PublicAPI(stability="beta")
```
Comment (amogkam) [inline python/ray/util/sgd/data/dataset.py:5]: This is actually Deprecated fully and replaced by Rat Data

**Comment by krfricke** [comment]
Thanks, take another look? :-)

**Review by amogkam — APPROVED** [review]
Thanks!

**Pull request merged**